rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -47,8 +47,23 @@ func resourceRancher2MachineConfigV2Create(d *schema.ResourceData, meta interfac
 		MinTimeout: 3 * time.Second,
 	}
 	_, waitErr := stateConf.WaitForState()
+
+	// Terraform may finish creating the machine config v2 before rancher has set get
+	// permissions on the resource. Retry a few times to see if the state will return
+	// without error.
 	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
+		if IsNotAccessibleByID(waitErr) {
+			var numberOfRetries = 0
+			for numberOfRetries < 3 {
+				_, err = stateConf.WaitForState()
+				if err != nil {
+					numberOfRetries++
+					continue
+				}
+				return flattenMachineConfigV2(d, newObj)
```

Comment:
```suggestion
			for i := 0; i < 3; i++ {
				_, err = stateConf.WaitForState()
				if err != nil {
					continue
				}
				return flattenMachineConfigV2(d, newObj)
```
I feel like we can pull the loop counter into the loop scope and give it a simpler name, otherwise logic looks good.